nscplugin/src/main/scala/scala/scalanative/nscplugin/NirGenExpr.scala

```diff
@@ -1277,7 +1279,8 @@ trait NirGenExpr { self: NirGenPhase =>
                               _))))))),
             _),
             _) =>
-          val chars = Val.Chars(str.replace("\\n", "\n").replace("\\r", "\r"))
+
+          val chars = Val.Chars(treatEscapes(str))
```

Comment:
Is there any reason why you're not using the built-in `StringContext.treatEscapes(str)`?

Reply:
It would throw exceptions for unknown escapes from C, that's why it could not be used.

Reply:
Why would we want to support escapes from C? This is a Scala string interpolation, it should respect Scala's string interpolation rules.

Reply:
I've changed it to use `StringContext.processEscapes` as mentioned earlier